Python/Product/TestAdapter.Executor/Pytest/TestResultParser.cs

```diff
@@ -23,69 +23,61 @@
 
 namespace Microsoft.PythonTools.TestAdapter.Pytest {
     public class TestResultParser {
-
-
         /// <summary>
-        /// Parses the junit xml for test results and matches them to the corresponding original TestCase using 
+        /// Parses the junit xml for test results and matches them to the corresponding original vsTestCase using 
         /// </summary>
         /// <param name="junitXmlPath"></param>
         /// <param name="tests"></param>
         /// <returns></returns>
-        public static IEnumerable<TestResult> Parse(string junitXmlPath, IEnumerable<TestCase> tests) {
-            var vsTestResults = tests.Select(t => new TestResult(t) { Outcome = TestOutcome.NotFound }).ToList();
+        internal static string GetPytestId(XPathNavigator result) {
+            var file = result.GetAttribute("file", "");
+            var funcname = result.GetAttribute("name", "");
+            var classname = result.GetAttribute("classname", "");
+            var line = result.GetAttribute("line", "");
 
-            if (!File.Exists(junitXmlPath)) {
-                return vsTestResults;
+            if (IsPytestResultValid(file, classname, funcname, line)) {
+                var pytestId = CreatePytestId(file, classname, funcname);
+                return pytestId;
             }
+            return null;
+        }
 
-            var doc = Read(junitXmlPath);
-            XPathNodeIterator xmlTestCases = doc.CreateNavigator().Select("/testsuite/testcase");
-
-            foreach (XPathNavigator pytestcase in xmlTestCases) {
-                var file = pytestcase.GetAttribute("file", "");
-                var name = pytestcase.GetAttribute("name", "");
-                var classname = pytestcase.GetAttribute("classname", "");
-
-                if (String.IsNullOrEmpty(file) ||
-                    String.IsNullOrEmpty(name) ||
-                    String.IsNullOrEmpty(classname) ||
-                    !Int32.TryParse(pytestcase.GetAttribute("line", String.Empty), out int line))
-                {
-                    var message = String.Empty;
-                    if(pytestcase.HasChildren) {
-                        pytestcase.MoveToFirstChild();
-                        message = pytestcase.GetAttribute("message", String.Empty) + pytestcase.Value;
-                    }
-                    Debug.WriteLine("Test result parse failed: {0}".FormatInvariant(message));
-                    continue;
-                }
-
-                // Match on classname and function name for now
-                var result = vsTestResults
-                    .Where(tr =>
-                    String.Compare(tr.TestCase.DisplayName, name, StringComparison.InvariantCultureIgnoreCase) == 0 &&
-                    String.Compare(tr.TestCase.GetPropertyValue<string>(Pytest.Constants.PyTestXmlClassNameProperty, default(string)), classname, StringComparison.InvariantCultureIgnoreCase) == 0) 
-                    .FirstOrDefault();
-
```

Comment:
Previously we did a linear double string compare search to match xml pytest results. Now on the right side we do this lookup in the caller TestExecutorPytest.cs and use a dictionary to map pytestId to TestResults for faster lookup using CreatePytestIdToVsTestResultsMap(..)